src/services/combineResponses.ts

```diff
@@ -11,18 +11,56 @@ import {
 
 import { logger } from './logger';
 
+function getFrameKey(frame: DataFrame): string {
+  const field = frame.fields.find((f) => f.type === FieldType.number);
+  if (!field) {
+    throw new Error(`Unable to find number field on sharded dataframe!`);
+  }
+
+  if (!frame.name) {
+    if (field.labels) {
+      frame.name = (frame.refId ?? '') + JSON.stringify(field.labels);
+    } else {
+      if (!frame.refId) {
+        throw new Error('Unable to find refId field on sharded dataframe!');
+      }
+      frame.name = frame.refId;
+    }
+  }
+  return frame.name;
+}
+
 export function combineResponses(currentResult: DataQueryResponse | null, newResult: DataQueryResponse) {
   if (!currentResult) {
     return cloneQueryResponse(newResult);
   }
 
-  newResult.data.forEach((newFrame) => {
-    const currentFrame = currentResult.data.find((frame) => shouldCombine(frame, newFrame));
-    if (!currentFrame) {
-      currentResult.data.push(cloneDataFrame(newFrame));
-      return;
+  const currentResultLabelsMap = new Map<string, DataFrame>();
+  currentResult.data.forEach((frame: DataFrame) => {
+    currentResultLabelsMap.set(getFrameKey(frame), frame);
+  });
+
+  newResult.data.forEach((newFrame: DataFrame) => {
+    let currentFrame: DataFrame | undefined = undefined;
+    const frameType = newFrame.meta?.type;
+    if (frameType === DataFrameType.TimeSeriesMulti) {
+      const key = getFrameKey(newFrame);
+
+      if (currentResultLabelsMap.has(key)) {
+        currentFrame = currentResultLabelsMap.get(key);
+        mergeFrames(currentFrame!, newFrame);
+      } else {
+        currentResult.data.push(cloneDataFrame(newFrame));
+      }
+    } else {
+      console.log('fallback merge');
```

Comment:
I think this can get removed, it's never been called in my testing, new dataplane spec or old

Reply:
Or just leave 

```
currentResult.data.push(cloneDataFrame(newFrame));
```

Reply:
You want, but the user probably doesn't.

Reply:
we're catching the error and logging through faro, user shouldn't be aware